packages-next/admin-ui/src/system/createAdminUIServer.ts

```diff
@@ -2,16 +2,16 @@ import Path from 'path';
 import url from 'url';
 import next from 'next';
 import express from 'express';
-import type { KeystoneSystem } from '@keystone-next/types';
+import type { KeystoneSystem, KeystoneConfig } from '@keystone-next/types';
 
 const dev = process.env.NODE_ENV !== 'production';
 
-export const createAdminUIServer = async (system: KeystoneSystem) => {
+export const createAdminUIServer = async (ui: KeystoneConfig['ui'], system: KeystoneSystem) => {
   const app = next({ dev, dir: Path.join(process.cwd(), '.keystone', 'admin') });
   const handle = app.getRequestHandler();
   await app.prepare();
 
-  const publicPages = system.config.ui?.publicPages ?? [];
+  const publicPages = ui?.publicPages ?? [];
```

Comment:
I find this variable to be really obscure. First suggestion is pass `config`, but I guess you'd prefer to be clearer that we don't need the whole config.

So maybe call it `uiConfig` instead?

Reply:
The aim of the change is to make it clear that we're not consuming the entire `config` object here.

I would think that this is where our types come to the rescue? It's clearly of type `KeystoneConfig['ui']`, and by using a consistent variable name we can mentally map it directly onto the original `config` object. I find that keeping consistent variables names for the same thing throughout the system is a bigger payoff.

Reply:
I... guess? care factor about 2/5 here. Just stood out to me.

> The aim of the change is to make it clear that we're not consuming the entire config object here.

Yep saw that one coming, and agree 🙂 

> I would think that this is where our types come to the rescue?

At the top of the function, for sure, I just feel weirdly uncertain lower down seeing `ui` without anything to say "this is part of the config". Maybe because I work with a lot more "UI" code so it's a super ambiguous variable name for me?

> by using a consistent variable name we can mentally map it directly onto the original config object

I _think_ I agree with this, although so many other places this is not an assumption I'd ever feel safe makeing.

e.g if it were `lists` being passed in, I'd also say "let's call that `listsConfig` because otherwise, is this `keystone.lists` or `adminMeta.lists` or `context.lists` etc.

So I think I'd prefer to introduce a convention like that: if you slice out just part of config to pass around, call it `${key}Config`. Elsewhere we might have `listsConfig`, `serverConfig`, `dbConfig` etc.

> I find that keeping consistent variables names for the same thing throughout the system is a bigger payoff.

💯 :this: